pysb/importers/sbml.py

```diff
@@ -0,0 +1,129 @@
+           query the pathway commons web service.
+    :param verbose: Print the SBML conversion output to the console if True
+    :return: BNGL output filename
+    """
+    sbmltrans_bin = os.path.join(os.path.dirname(_get_bng_path()),
+                                 'bin/sbmlTranslator')
+    sbmltrans_args = [sbmltrans_bin, '-i', input_file]
+    if output_file is None:
+        output_file = os.path.splitext(input_file)[0] + '.bngl'
+    sbmltrans_args.extend(['-o', output_file])
+
+    if convention_file:
+        sbmltrans_args.extend(['-c', convention_file])
+
+    if naming_conventions:
+        sbmltrans_args.extend(['-n', naming_conventions])
+
+    if user_structures:
+        sbmltrans_args.extend(['-u', user_structures])
+
+    if molecule_id:
+        sbmltrans_args.append('-id')
+
+    if atomize:
+        sbmltrans_args.append('-a')
+
+    if pathway_commons:
+        sbmltrans_args.append('-p')
+
+    if verbose:
+        print("sbmlTranslator command:")
+        print(" ".join(sbmltrans_args))
+
+    p = subprocess.Popen(sbmltrans_args,
+                         cwd=os.getcwd(),
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE)
+
+    if verbose:
+        for line in iter(p.stdout.readline, b''):
+            print(line, end="")
+    (p_out, p_err) = p.communicate()
+    if p.returncode:
+        raise SbmlTranslationError(p_out.decode('utf-8') + "\n" +
+                                   p_err.decode('utf-8'))
+
+    return output_file
+
+
+def model_from_sbml(filename, cleanup=True, **kwargs):
+    """
+    Create a PySB Model object from an Systems Biology Markup Language (SBML)
+    file, using BioNetGen's
+    `sbmlTranslator <http://bionetgen.org/index.php/SBML2BNGL>`_,
+    which can attempt to extrapolate higher-level (rule-based) structure
+    from an SBML source file (argument atomize=True). The model is first
+    converted into BioNetGen language by sbmlTranslator, then PySB's
+    :class:`BnglBuilder` class converts the BioNetGen language model into a
+    PySB Model.
+
+    Limitations
+    -----------
+
+    Read the `sbmlTranslator documentation
+    <http://bionetgen.org/index.php/SBML2BNGL>`_ for further information on
+    sbmlTranslator's limitations.
+
+    Parameters
+    ----------
+    filename :
+        A Systems Biology Markup Language .sbml file
+    cleanup : bool
+        Delete temporary directory on completion if True. Set to False for
+        debugging purposes.
+    **kwargs: kwargs
+        Keyword arguments to pass on to :func:`SbmlBuilder.sbml_translator`
+    """
+    tmpdir = tempfile.mkdtemp()
+    verbose = kwargs.get('verbose', False)
+    if verbose:
+        print("Performing SBML to BNGL translation in temporary "
+              "directory %s" % tmpdir)
+    try:
+        bngl_file = os.path.join(tmpdir, 'model.bngl')
+        sbml_translator(filename, bngl_file, **kwargs)
+        return model_from_bngl(bngl_file)
+    finally:
```

Comment:
In this case, no exceptions are caught; they are handled as if the `try`/`finally` block wasn't there. Catching and suppressing exceptions would require an `except: pass` block.

As an example, the following raises an Exception:

```python
try:
    raise Exception()
finally:
    print('finally')
```